### PR TITLE
Use a custom Docker image to speed up Circle CI

### PR DESCRIPTION
**Pull request by Luthaf**
A lot of time (~ 10min) is spent installing dependencies on Circle CI. This PR creates a custom docker image with dependencies pre-installed to try to reduce the build time. 

~~Unfortunately, I hit the error from #127 when building with GCC 9, so this will have to be fixed first.~~

**Unresolved:**

- [x] The docker image is currently pushed as luthaf/rascal-ci. Should we setup a docker hub account for rascal, independent if myself? Set-up as cosmoepfl/rascal-ci

### CODE REVIEW

**Comment by felixmusil** [comment]
Having a main image is indeed much better. 
Why is GCC 9 important to this PR ?

**Comment by Luthaf** [comment]
>  Why is GCC 9 important to this PR ?

Not really important by itself, I also wanted to improve build time by using only selected compilers, as discussed in the last meeting. But if I can't fix the issue, I'll fallback to gcc 7

File: .circleci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+RUN apt-get update && \
+    apt-get install -y software-properties-common wget && \
+    wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -  && \
+    add-apt-repository ppa:ubuntu-toolchain-r/test && \
+    add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+
+RUN apt-get update && \
+    apt-get install -y \
+        git \
+        cmake \
```
Comment (felixmusil) [inline .circleci/Dockerfile]:
```suggestion
        cmake \
        vim \
```

Comment (felixmusil) [reply .circleci/Dockerfile]: in the case one would need to interact with files on the image

**Comment by Luthaf** [comment]
> A lot of time (~ 10min) is spent installing dependencies on Circle CI. This PR creates a custom docker image with dependencies pre-installed to try to reduce the build time.

After more checking, the 10min "installing dependencies" step was an outlier, and the usual timing for this step is more 2min. This makes this PR less interesting, since it introduces more complexity for less gains.

**Comment by felixmusil** [comment]
Still it makes more sense to download and install the common utilities beforehand so I would still move forward with this PR.

File: .circleci/Dockerfile (hunk 1 above)
Comment (Luthaf) [reply .circleci/Dockerfile]: sounds good!

**Comment by Luthaf** [comment]
> The docker image is currently pushed as luthaf/rascal-ci. Should we setup a docker hub account for rascal, independent if myself?

@ceriottm what do you think about this?

**Pull request merged**